plugins/cue_points/quiz/lib/model/kQuizManager.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * @package plugins.quiz
+ */
+class kQuizManager implements kObjectAddedEventConsumer, kObjectChangedEventConsumer
```

Comment:
It's important, once because it will simplify the code significantly, but mostly because the use of CuePoint::isCustomDataModified is wrong and won't catch all scenarios, I can explain more offline.